### PR TITLE
Add concept names

### PR DESCRIPTION
**Pull request by Kadrian**

### CODE REVIEW

File: frontend/src/js/entity-history/timeline/ConceptName.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  datasetId: DatasetT["id"] | null;
+  row: EntityEvent;
+  column: ColumnDescription;
+  rootConceptId: ConceptIdT;
+}
+
+const ConceptName = ({ datasetId, row, column, rootConceptId }: Props) => {
+  // TODO: refactor. It's very implicit that the id is
+  // somehow containing the datasetId.
+  if (!datasetId) return null;
```
Comment (awildturtok) [inline frontend/src/js/entity-history/timeline/ConceptName.tsx:22]: where are you missing the datasetid? And where are you relying on it?

File: frontend/src/js/entity-history/timeline/ConceptName.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  if (fullConceptId === rootConceptId) {
+    return conceptName;
+  }
+
+  const rootConcept = getConceptById(rootConceptId, rootConceptId);
+
+  return (
+    <div>
+      {rootConcept ? `${rootConcept.label} ` : null}
```
Comment (awildturtok) [inline frontend/src/js/entity-history/timeline/ConceptName.tsx:49]: The screenshot you posted earlier had the concept label standard and the element bold. Imo bolding it is not necessary and it looks a bit unbalanced, it's probably fine going standard for both?

**Review by awildturtok — APPROVED** [review]

File: frontend/src/js/concept-trees/globalTreeStoreHelper.ts [hunk 3]
```diff
@@ -47,11 +47,18 @@ export function setTree(
...
+): ConceptT | null {
+  if (rootId) {
+    return window.conceptTrees[rootId][conceptId] || null;
+  }
+
+  const conceptTreeRootIds: ConceptIdT[] = Object.keys(window.conceptTrees);
 
-  for (let i = 0; i < keys.length; i++) {
-    const concept = window.conceptTrees[keys[i]][conceptId];
+  for (const rootId of conceptTreeRootIds) {
```
[the remark this answers is not in the document]
Comment (Kadrian) [reply frontend/src/js/concept-trees/globalTreeStoreHelper.ts:60]: This was O(n), with n = number of all root concepts, so not too bad. Now it's instant time whenever a root id is available

File: frontend/src/js/entity-history/timeline/ConceptName.tsx (hunk 1 above)
Comment (Kadrian) [reply frontend/src/js/entity-history/timeline/ConceptName.tsx:22]: The dataset id is available as soon as datasets have been loaded (it's not defined before that) – and now I need it here to create the full concept ID, I'd argue the backend should send the full ID as part of the csv, so including the dataset id

File: frontend/src/js/entity-history/timeline/ConceptName.tsx (hunk 2 above)
Comment (Kadrian) [reply frontend/src/js/entity-history/timeline/ConceptName.tsx:49]: yes, this is peliminary, will adjust in a follow-up

**Pull request merged**